python-lib/spell_checker/text_preprocessing.py

```diff
@@ -0,0 +1,101 @@
+import re
+import spacy.lang
+
+# All languages available in Spacy and for which we have a frequency dictionnary
+lang_dict = {
+    "sq": "Albanian",
+    "ar": "Arabic",
+    "bg": "Bulgarian",
+    "ca": "Catalan",
+    "zh": "Chinese",
+    "hr": "Croatian",
+    "cs": "Czech",
+    "da": "Danish",
+    "nl": "Dutch",
+    "en": "English",
+    "et": "Estonian",
+    "fi": "Finnish",
+    "fr": "French",
+    "de": "German",
+    "el": "Greek",
+    "he": "Hebrew",
+    "hu": "Hungarian",
+    "is": "Icelandic",
+    "id": "Indonesian",
+    "it": "Italian",
+    "ja": "Japanese",
+    "ko": "Korean",
+    "lv": "Latvian",
+    "lt": "Lithuanian",
+    "fa": "Persian",
+    "pl": "Polish",
+    "pt": "Portuguese",
+    "ro": "Romanian",
+    "ru": "Russian",
+    "sr": "Serbian",
+    "sk": "Slovak",
+    "sl": "Slovenian",
+    "es": "Spanish",
+    "sv": "Swedish",
+    "th": "Thai",
+    "tr": "Turkish",
+    "uk": "Ukrainian",
+    "vi": "Vietnamese",
+}
+
+
+class PreprocessText:
+    def __init__(self, all_languages, params=None):
+        self.all_languages = all_languages
+        if params != None:
+            self.params = params
+
+    def _get_all_tokenizers(self):
+
+        language_modules = {}
+        nlps = {}
+        tokenizers = {}
+
+        for lang_code, lang_name in lang_dict.items():
+
+            if lang_code not in self.all_languages:
```

Comment:
We should logging.warn if the language code is not supported, and make sure the output is empty